### PR TITLE
P4 2700 basm api client

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Changes:

The changes in this PR replace the existing CSV based location name lookup with calls the BaSM backend API instead.

No authentication is required when calling this endpoint as it does not require authentication.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt [hunk 1]
```diff
@@ -118,9 +118,8 @@ class SecurityConfiguration<S : Session> : WebSecurityConfigurerAdapter() {
     }
   }
 
-  // Needed for Thymeleaf security extras.
   @Bean
-  fun securityDialect(): SpringSecurityDialect? {
+  fun securityDialectForThymeleafSecurityExtras(): SpringSecurityDialect? {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt:122]: Unrelated change.  Renamed method so can remove comment.

File: .env [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
-BASM_BUCKET_NAME=basm
-JPC_BUCKET_NAME=jpc
 AWS_ACCESS_KEY_ID=foo
 AWS_SECRET_ACCESS_KEY=bar
 AWS_DEFAULT_REGION=eu-west-2
-S3_ENDPOINT=http://localhost:4572
-IMPORT_FILES_LOCATIONS=schedule_34_locations_sample.xlsx
-IMPORT_FILES_PRICES_SERCO=serco_sample.xlsx
-IMPORT_FILES_PRICES_GEO=geoamey_sample.xlsx
+BASM_API_BASE_URL=http://localhost:9999
```
Comment (matt-crowson-moj) [inline .env:5]: This is the new addition.  This is pointing at WireMock that starts up via the docker-compose (done in a separate PR).

File: .env [hunk 3]
```diff
@@ -1,17 +1,18 @@
 APP_DB_URL="jdbc:postgresql://127.0.0.1:5433/jpc?reWriteBatchedInserts=true&user=jpc&password=letmein"
-BASM_BUCKET_NAME=basm
```
Comment (matt-crowson-moj) [inline .env:2]: Have put the env vars in alphabetical order just to make reading easier.

**Review by Thource — APPROVED** [review]
